api/src/utils/put-utils.js

```diff
@@ -112,3 +114,122 @@ exports.getDotNotation = function(obj, target, prefix) {
 
   return target;
 };
+
+exports.editRecordWithFlavours = async function (args, res, next, incomingObj, editMaster, PostFunctions, masterSchemaName, flavourFunctions = {}) {
+  let flavours = [];
+  let flavourIds = [];
+  let promises = [];
+
+  // make a copy of the incoming object for use by the flavours only
+  const flavourIncomingObj = { ...incomingObj };
+  // Remove fields that should not be inherited from the master record
+  delete flavourIncomingObj._id;
+  delete flavourIncomingObj._schemaName;
+  delete flavourIncomingObj._flavourRecords;
+  delete flavourIncomingObj._master;
+  delete flavourIncomingObj.read;
+  delete flavourIncomingObj.write;
+
+  // Prepare flavours
+  const entries = Object.entries(flavourFunctions);
+  // Example of entries: [['OrderLNG', createLNG()], ['OrderNRCED', createNRCED()]]
+
+  if (entries.length > 0 && Object.keys(flavourIncomingObj).length > 0) {
```

Comment:
This function, and the similar one in the post utils, are prime candidates from some cost effective unit tests.  With all of the nested conditionals and loops, they would be difficult for a new dev to debug/update if they had to.